rpcs3/rpcs3qt/game_list_frame.cpp

```diff
@@ -284,6 +284,99 @@ bool game_list_frame::IsEntryVisible(const game_info& game, bool search_fallback
 	return is_visible && matches_category() && SearchMatchesApp(qstr(game->info.name), serial, search_fallback);
 }
 
+bool game_list_frame::RemoveContentPath(const std::string& path, const std::string& desc)
+{
+	if (fs::exists(path))
+	{
+		if (fs::is_dir(path))
+		{
+			if (fs::remove_all(path))
+			{
+				game_list_log.notice("Removed '%s' directory: '%s'", desc, path);
+			}
+			else
+			{
+				game_list_log.error("Could not remove '%s' directory: '%s' (%s)", desc, path, fs::g_tls_error);
+
+				return false;
+			}
+		}
+		else // If file
+		{
+			if (fs::remove_file(path))
+			{
+				game_list_log.notice("Removed '%s' file: '%s'", desc, path);
+			}
+			else
+			{
+				game_list_log.error("Could not remove '%s' file: '%s' (%s)", desc, path, fs::g_tls_error);
+
+				return false;
+			}
+		}
+	}
+
+	return true;
+}
+
+bool game_list_frame::RemoveContentBySerial(const std::string& base_dir, const std::string& serial, const std::string& desc)
+{
+	bool success = true;
+
+	for (const auto& entry : fs::dir(base_dir))
+	{
+		// Search for any path starting with serial (e.g. BCES01118_BCES01118)
+		bool found = entry.name.starts_with(serial);
+
+		if (found)
+		{
+			if (entry.is_directory)
+			{
+				if (fs::remove_all(base_dir + entry.name))
+				{
+					game_list_log.notice("Removed '%s' directory: '%s'", desc, base_dir + entry.name);
+				}
+				else
+				{
+					game_list_log.error("Could not remove '%s' directory: '%s' (%s)", desc, base_dir + entry.name, fs::g_tls_error);
+
+					success = false;
+				}
+			}
+			else // If file
+			{
+				if (fs::remove_file(base_dir + entry.name))
+				{
+					game_list_log.notice("Removed '%s' file: '%s'", desc, base_dir + entry.name);
+				}
+				else
+				{
+					game_list_log.error("Could not remove '%s' file: '%s' (%s)", desc, base_dir + entry.name, fs::g_tls_error);
+
+					success = false;
+				}
+			}
+		}
```

Comment:
You literally wrote this in the function above.
Just do
```suggestion
		if (!RemoveContentPath(base_dir + entry.name, desc))
		{
			success = false;
		}
```